Sources/MessagingPush/MessagingPush.swift

```diff
@@ -50,6 +53,8 @@ public class MessagingPush: ModuleTopLevelObject<MessagingPushInstance>, Messagi
     @available(iOSApplicationExtension, unavailable)
     public static func initialize(withConfig config: MessagingPushConfigOptions = MessagingPushConfigBuilder().build()) -> MessagingPushInstance {
         shared.initializeModuleIfNotAlready {
+            // set moduleConfig before creating implementation instance as dependencies inside instance may directly use moduleConfig from MessagingPush.
+            Self.moduleConfig = config
```

Comment:
Thanks for adding these comments! If the order matters, a comment is good to have!

Reply:
Since all modules rely on the same order, that is Config gets built first and then implemented, maybe we should have a structure in place so that whenever a new module is built it follows a strict protocol of config and implementation.